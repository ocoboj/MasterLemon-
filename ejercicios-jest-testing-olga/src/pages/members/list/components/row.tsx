import * as React from 'react';
import { Member } from '../viewModel';

interface Props {
  member: Member;
}

export const Row: React.StatelessComponent<Props> = (props) => (
  <tr>
    <td>
      <img src={props.member.avatarUrl} style={{ maxWidth: '10rem' }} />
    </td>
    <td>
      <span>{props.member.id}</span>
    </td>
    <td>
      <span>{props.member.name}</span>
    </td>
  </tr>
);
