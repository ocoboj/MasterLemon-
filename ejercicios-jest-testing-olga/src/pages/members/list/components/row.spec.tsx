import { shallow } from "enzyme";
import { Row } from "./row";
import * as React from 'react'

describe('pages/member/list/components/ row specs', () => {
    it('should render as expected passing required properties', () => {
        const props = {
            member: {
                id: 3,
                name: 'name test',
                avatarUrl: 'url test',
            }
        }

        const component = shallow(
            <Row {...props}/>
        );

        expect(component).toMatchSnapshot();
    });
});