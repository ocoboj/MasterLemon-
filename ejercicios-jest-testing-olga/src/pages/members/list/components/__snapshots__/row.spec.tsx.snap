// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/member/list/components/ row specs should render as expected passing required properties 1`] = `
<tr>
  <td>
    <img
      src="url test"
      style={
        Object {
          "maxWidth": "10rem",
        }
      }
    />
  </td>
  <td>
    <span>
      3
    </span>
  </td>
  <td>
    <span>
      name test
    </span>
  </td>
</tr>
`;
